### DISCUSSION TITLE
Using an external MinIO, the key cannot be found

### DISCUSSION THREAD

**Discussion by huangzgx** [milvus-io/milvus#28064] [Q&A and General discussion]
How can I use an existing MinIO instance when deploying a standalone version of Milvus using a docker-compose.yml file? I have modified the YAML file but after deploying, it prompts that the key cannot be found. Please assist me with this issue

`version: '3.5'

services:
  etcd:
    container_name: milvus-etcd
    image: quay.io/coreos/etcd:v3.5.5
    restart: always
    privileged: true
    network_mode: host
    environment:
      - ETCD_AUTO_COMPACTION_MODE=revision
      - ETCD_AUTO_COMPACTION_RETENTION=1000
      - ETCD_QUOTA_BACKEND_BYTES=4294967296
      - ETCD_SNAPSHOT_COUNT=50000
    volumes:
      - ${DOCKER_VOLUME_DIRECTORY:-.}/volumes/etcd:/etcd
    command: etcd -advertise-client-urls=http://127.0.0.1:2379 -listen-client-urls http://0.0.0.0:2379 --data-dir /etcd
    healthcheck:
      test: ["CMD", "etcdctl", "endpoint", "health"]
      interval: 30s
      timeout: 20s
      retries: 3

  standalone:
    container_name: milvus-standalone
    image: milvusdb/milvus:v2.3.1
    command: ["milvus", "run", "standalone"]
    restart: always
    privileged: true
    network_mode: host
    security_opt:
    - seccomp:unconfined
    environment:
      ETCD_ENDPOINTS: 10.0.12.11:2379
      MINIO_ADDRESS: 10.0.12.11:9000
      MINIO_ACCESS_KEY: admin
      MINIO_SECRET_KEY: 5v2gENeC4P
    volumes:
      - ${DOCKER_VOLUME_DIRECTORY:-.}/volumes/milvus:/var/lib/milvus
    healthcheck:
      test: ["CMD", "curl", "-f", "http://localhost:9091/healthz"]
      interval: 30s
      start_period: 90s
      timeout: 20s
      retries: 3
    depends_on:
      - "etcd"
`

**Comment by yhmo** [comment]
Milvus uses minio as obj storage and uses etcd as meta store. By reading the meta in etcd, milvus knows where to find a collection's data files in minio. If the etcd service is recreated, the metadata is lost and the new cluster will not be able to know how many collections there and where they were persisted.

Theoretically, if the previous etcd is using this folder as datapath:  `${DOCKER_VOLUME_DIRECTORY:-.}/volumes/etcd:/etcd`, now you reuse the etcd path for the new milvus service, it should work well.
